### DISCUSSION TITLE
How Emoji/Role cache works?

### DISCUSSION THREAD

**Discussion by FerroEduardo** [discord-jda/JDA#2146] [Questions and Help] [answered]
I tried to find information about **how the cache is filled**, but could not find details.  
Using a `SlashCommandEvent` as example, when I call `event.getGuild().getEmotes()` or `event.getGuild().getRoles()`, the cache list is refresh every event call with updated elements, or it is refreshed somewhere else?

**Comment by MinnDevelopment** [comment, marked as answer]
The cache is updated by other events such as:

- GUILD_ROLE_CREATE
- GUILD_ROLE_UPDATE
- GUILD_ROLE_DELETE
- GUILD_EMOJIS_UPDATE

You can find the full list of events [here](https://discord.com/developers/docs/topics/gateway#commands-and-events). It is handled by the handler classes in [net.dv8tion.jda.internal.handle](https://github.com/DV8FromTheWorld/JDA/tree/master/src/main/java/net/dv8tion/jda/internal/handle).

**Reply by FerroEduardo** [reply]
Thanks a lot
